cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -188,8 +187,14 @@ public void create(HttpRequest request, HttpResponder responder,
     // Check for namespace existence. Throws NotFoundException if namespace doesn't exist
     namespaceClient.get(namespaceId);
 
-    // Verify stream name
-    Id.Stream streamId = Id.Stream.from(namespaceId, stream);
+    Id.Stream streamId;
+    try {
+      // Verify stream name
+      streamId = Id.Stream.from(namespaceId, stream);
+    } catch (IllegalArgumentException e) {
+      LOG.error("Failed to create stream {}", stream, e);
```

Comment:
Don't log error (in fact, no need to log). If really want to log, log as debug or trace. A user error shouldn't result in ERROR level log.